heroic-core/src/main/java/com/spotify/heroic/metric/LocalMetricManager.java

```diff
@@ -282,22 +306,32 @@ public QueryTrace buildTrace() {
                 }
 
                 final List<Callable<AsyncFuture<FetchData.Result>>> fetches = new ArrayList<>();
-
-                /* setup fetches */
                 accept(metricBackend -> {
                     for (final Series series : result.getSeries()) {
+                        final Span fetchSeries = tracer.spanBuilderWithExplicitParent(
```

Comment:
I think its a good pattern to end the spans in the same context where they're started. Closing it from the function where its passed in makes it easy to accidentally leak a span (especially if that child passes it further down the stack) and makes it harder to reason about what the span is actually measuring.

Reply:
👍 💯 - definitely that's a good best practice we should document.